### PR TITLE
Test for marshalling Organizations and make JSON comparison non sensitive to key order

### PR DESCRIPTION
**Pull request by sgarciac**
This pull request contains two items:

1. Modifies the JSON comparison utility for checking marshalling, making it non sensible to key order.This makes it easier to write tests for marshalling, since there is no need to worry about keeping the right order of the JSON keys in the wanted string.  It also removes a non working part of the test. *I made sure old tests using this utility would still break if given the wrong input*.
2. Adds a new test for marshalling organizations (issue #55 )

### CODE REVIEW

File: github/github_test.go [hunk 1]
```diff
@@ -135,32 +134,26 @@ func testBody(t *testing.T, r *http.Request, want string) {
 	}
 }
 
-// Helper function to test that a value is marshalled to JSON as expected.
+// Test whether the marshalling of v produces a JSON that corresponds
```
Comment (gmlewis) [inline github/github_test.go]: Let's please spell "marshal" and "unmarshal" consistently throughout the PR with a single "l".
Please also terminate the comment sentence with a period.

Comment (sgarciac) [reply github/github_test.go]: done

Comment (sgarciac) [reply github/github_test.go]: (also, added the issue I'm helping with to the description of the PR and the commit message)

File: github/github_test.go [hunk 2]
```diff
@@ -135,32 +134,26 @@ func testBody(t *testing.T, r *http.Request, want string) {
 	}
 }
 
-// Helper function to test that a value is marshalled to JSON as expected.
+// Test whether the marshaling of v produces a JSON that corresponds
```
Comment (gmlewis) [inline github/github_test.go]: Let's please say "produces JSON" instead of "produces a JSON".

File: github/github_test.go [hunk 3]
```diff
@@ -135,32 +134,26 @@ func testBody(t *testing.T, r *http.Request, want string) {
...
-	j, err := json.Marshal(v)
-	if err != nil {
-		t.Errorf("Unable to marshal JSON for %v", v)
+	// Unmarshal the wanted JSON, to verify its correctness, and marshal it back
+	// to sort the keys.
+	u := reflect.New(reflect.TypeOf(v)).Interface()
+	if err := json.Unmarshal([]byte(want), &u); err != nil {
+		t.Errorf("Unable to unmarshal JSON for %v: %v", want, err)
 	}
+	w, _ := json.Marshal(u)
```
Comment (gmlewis) [inline github/github_test.go]: Please test the returned `err` here instead of ignoring it.

File: github/github_test.go (hunk 2 above)
Comment (sgarciac) [reply github/github_test.go]: done!

File: github/github_test.go (hunk 3 above)
Comment (sgarciac) [reply github/github_test.go]: done! (however, I wonder if it's ever possible for the marshal here to fail, if the immediate unmarshal had no errors)

Comment (gmlewis) [reply github/github_test.go]: Probably not... but we would like to promote good Go programming practices for all developers who take a look at this code, whether they are new to Go or not... and if developers start ignoring error codes when they think there is no possible way that it could error, that opens up a potential world of pain further down the line.

Let's say, for example, we left it as-is, but then someone in the future decided to change the code above this that generated the value of `u`, but left this line the same. Suddenly, it might be possible that `u` was malformed, but this line would not catch that problem.

In summary, it is always safest to check returned error values in Go to minimize future negative surprises.  Thanks.

File: github/github_test.go [hunk 4]
```diff
@@ -135,32 +134,29 @@ func testBody(t *testing.T, r *http.Request, want string) {
...
+	// Unmarshal the wanted JSON, to verify its correctness, and marshal it back
+	// to sort the keys.
+	u := reflect.New(reflect.TypeOf(v)).Interface()
+	if err := json.Unmarshal([]byte(want), &u); err != nil {
+		t.Errorf("Unable to unmarshal JSON for %v: %v", want, err)
+	}
+	w, err := json.Marshal(u)
 	if err != nil {
-		t.Errorf("Unable to marshal JSON for %v", v)
+		t.Errorf("Unable to marshal JSON for %v", u)
```
Comment (gmlewis) [inline github/github_test.go]: nit: please change this from `%v` to `%#v` in order to better see the contents of `u`.

Comment (sgarciac) [reply github/github_test.go]: done!

**Comment by wesleimp** [comment]
👌
